packages/driver/src/cypress/runner.ts

```diff
@@ -1534,6 +1546,23 @@ export default {
           })
         }
 
+        // if either the TEST_BEFORE_RUN_EVENT or TEST_BEFORE_RUN_ASYNC_EVENT throws
+        // then override the test function to associate the error to the test
+        const handleBeforeTestEventError = (err: Error): boolean => {
+          const { fn } = runnable
+
+          const restore = () => {
+            return runnable.fn = fn
+          }
+
+          runnable.fn = () => {
+            restore()
```

Comment:
Could this line just be resetting runnable.fn to fn instead of calling the restore function?
```suggestion
            runnable.fn = fn
```